tests/unit/providers/test_anthropic_provider.py

```
@@ -122,6 +122,48 @@ async def test_completion_with_kwargs() -> None:
+    """An explicit timeout lifts the SDK pre-flight guard so the same large request reaches transport."""
+    provider = AnthropicProvider(api_key="sk-test")
+    params = CompletionParams(
+        model_id="claude-opus-5",
+        messages=[{"role": "user", "content": "Hi"}],
+        max_tokens=65536,
+        stream=False,
+    )
+
+    # The guard fires before the transport call, so reaching ``post`` proves it was skipped.
+    with patch.object(provider.client, "post", new=AsyncMock(side_effect=RuntimeError("reached transport"))):
+        with pytest.raises(RuntimeError, match="reached transport"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)